### PR TITLE
Add an example of dynamically showing labels

### PR DESCRIPTION
**Pull request by hyanwong**
This can be really useful for large trees

### CODE REVIEW

File: viz.md [hunk 1]
```diff
@@ -58,13 +60,23 @@ def viz_selection():
...
         population_size=1e4,
-        recombination_rate=1e-8,
+        recombination_rate=2e-8,
         sequence_length=sequence_length,
-        random_seed=9,
+        random_seed=222,
     )
+    tables = ts_selection.dump_tables()
+    tables.nodes.metadata_schema = tskit.MetadataSchema.permissive_json()
+    node_names = [{}] * tables.nodes.num_rows
```
Comment (jeromekelleher) [inline viz.md:73]: This is quite a complicated way to do it - why not:
```
tables.nodes.clear()
tables.nodes.metadata_schema = ...
for node in ts.nodes():
      metadata = {}
      if node.is_sample():
           metadata["name"] = ...
      tables.nodes.add_row(node.replace(metadata=metadata))
```
It won't be much slower (we're doing a bunch of stuff in Python either way) and it's a lot easier to follow.

**Review by jeromekelleher — APPROVED** [review]
LGTM - but suggest avoiding the packset metadata stuff, it's not sufficiently different in terms of performance to warrant the complexity.

File: viz.md (hunk 1 above)
Comment (hyanwong) [reply viz.md:73]: Sure, that's fine. I'm hoping most people won't be reading this bit of the code anyway (it's hidden, and there was a previous push motivated by Ben Haller to make all the example code at the end self-contained, so the reader doesn't need to look at the hidden stuff). As you say, it's best to make it clear to ourselves even in the hidden stuff though.

**Review by benjeffery — APPROVED** [review]
LGTM - agree with JK on the metadata though.

**Comment by hyanwong** [comment]
Done, ready to merge. Thanks for the comments, both: this is a nice capability to document.

**Comment by petrelharp** [comment]
This is *awesome*.

**Pull request merged**